tuplex/python/include/PythonDataSet.h

```diff
@@ -77,8 +77,9 @@ namespace tuplex {
         PythonDataSet resolve(const int64_t exceptionCode, const std::string& lambda_code, const std::string& pickled_code, const py::object& closure=py::object());
 
         py::object collect();
-        py::object take(const int64_t numRows);
+        py::object take(const int64_t topLimit, const int64_t bottomLimit);
```

Comment:
why not size_t here?

Reply:
We defined "take everything" in C++ dataset as having `topLimit` or `bottomLimit` equal to `size_t::max()`, but we can't do this in python since the python int technically does not have the maximum value. Therefore, the infinity is represented as -1 when passing from python to c++